src/content/docs/en/guides/cms/umbraco.mdx

```diff
@@ -0,0 +1,250 @@
+
+Once Astro has been setup and Umbraco installed, you are now able to create a blog that uses Umbraco as the CMS.
+
+### Creating the blog posts in Umbraco
+
+From the Umbraco backoffice, create a Document Type for a simple blog article called 'Article'.
+
+To follow along below, your 'Article' Document Type should have the following properties:
+
+- Title (DataType: Textstring)
+- Article Date (DataType: Date Picker)
+- Content (DataType: Richtext Editor)
+
+For this simple example, on the 'Article' document type, set the permission "Allow as root" to `true`.
+
+Please create and publish some test articles.
+
+Once these articles are published you should be able to retrieve the article details for all published articles as a JSON object via a GET request to the following URL:
+
+`http://localhost:44335/umbraco/delivery/api/v2/content?filter=contentType:article`
+
+Please replace the port number with your local development environment's HTTP endpoint port number.
+
+For information on creating Document Types, see [this guide](https://www.youtube.com/watch?v=otRuIkN80qM).
+
+### Displaying a list of blog posts in Astro
+
+Within our Astro project, this is the folder structure we will be using to create our blog:
+
+<FileTree>
+- src/
+  - layouts/
+    - Layout.astro
+  - pages/
+    - index.astro
+    - [...slug].astro
+</FileTree>
+
+If using the `Empty` project, please create the `layouts/` folder, and `Layout.astro` and `[...slug].astro` files.
+
+For more information on Layouts see the [Astro documentation](https://docs.astro.build/en/basics/layouts/).
+
+Add the following markup to the `Layout.astro` file:
+
+```astro title="src/layouts/Layout.astro"
+---
+---
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <title>My Blog with Astro and Umbraco</title>
+</head>
+<body>
+    <main>
+        <slot />
+    </main>
+</body>
+</html>
+```
+
+We will use the homepage `index.astro` file at the root to list out the blog articles.
+
+For this we will use a `fetch` to call the Content Delivery API `content` endpoint and filter by contentType of 'article'.
+
+The article objects will include the properties and content set in the CMS.
+
+We can then loop through the articles and display a desired listing with a link to the article.
+
+Replace the default contents of `index.astro` with the below:
+
+```astro title="src/pages/index.astro"
+---
+import Layout from '../layouts/Layout.astro';
+const res = await fetch('http://localhost/umbraco/delivery/api/v2/content?filter=contentType:article');
+const articles = await res.json();
+---
+<Layout>
+	<h2>Blog Articles</h2>
+	{
+        articles.items.map((article: any) => (
+            <div>
+                <h3>{article.properties.title}</h3>
+                <p>{article.properties.articleDate}</p>
+                <a href={article.route.path}>Read more</a>
+            </div>
+        ))
+    }
+</Layout>
+```
```

Comment:
``````suggestion
To create a list of blog posts, first `fetch` to call the Content Delivery API  `content` endpoint and filter by contentType of 'article'. The article objects will include the properties and content set in the CMS. You can then loop through the articles and display a each title with a link to its article.

Replace the default contents of `index.astro` with the following code:

```astro title="src/pages/index.astro"
---
import Layout from '../layouts/Layout.astro';
const res = await fetch('http://localhost/umbraco/delivery/api/v2/content?filter=contentType:article');
const articles = await res.json();
---
<Layout>
	<h2>Blog Articles</h2>
	{
        articles.items.map((article: any) => (
            <div>
                <h3>{article.properties.title}</h3>
                <p>{article.properties.articleDate}</p>
                <a href={article.route.path}>Read more</a>
            </div>
        ))
    }
</Layout>
```
``````